script.service.kodi.callbacks/resources/lib/utils/poutil.py

```diff
@@ -304,38 +608,115 @@ def getFileList(self):
         return files_to_scan
 
     def scanPyFilesForStrings(self):
+        """
+        Scans all .py files in directory tree for localized strings
+        :return: List of strings needing localization
+        :rtype: list
+        """
         files = self.getFileList()
         lstrings = []
         for myfile in files:
-            finds = []
-            with open(myfile, 'r') as f:
-                lines = ''.join(f.readlines())
+            with codecs.open(myfile, 'r', 'utf-8') as f:
+                lines = u''.join(f.readlines())
             try:
                 finds = self.find_localizer.findall(lines)
             except re.error:
-                pass
+                finds = []
             finally:
                 if len(finds) != 1:
-                    log(msg='Skipping file: %s, localizer not found' % myfile)
+                    log(msg=u'Skipping file: %s, localizer not found' % myfile)
                 else:
-                    findstr = r"%s\('(.+?)'\)" % finds[0]
+                    findstr = ur"%s\('(.+?)'\)" % finds[0]
                     find = re.compile(findstr)
-                    finds = []
                     try:
                         finds = find.findall(lines)
                     except re.error:
-                        pass
+                        finds = []
                     lstrings += finds
         return lstrings
 
     def updateStringsPo(self):
+        """
+        Scans files for localized strings and updates po
+        :return:
+        :rtype: None
+        """
         lstrings = self.scanPyFilesForStrings()
         for s in lstrings:
-            found, strid = self.podict.has_msgid(s)
+            found, strid = self.podict.has_msgid(s.decode('unicode_escape'))
             if found is False:
                 self.podict.addentry(strid, s)
         self.podict.write_to_file(self.current_working_English_strings_po)
 
-kodipo = KodiPo()
-_ = kodipo.getLocalizedString
-__ = kodipo.getLocalizedStringId
+
+def getPlatform():
```

Comment:
Yes - I run a 'build' routine outside of Kodi that utilizes this module. Still good for me to know about an alternative way to get that info.